diff --git a/vendors/wodotexteditor/Tools.js b/vendors/wodotexteditor/Tools.js
index b58f0eb..9cb7351 100644
--- a/vendors/wodotexteditor/Tools.js
+++ b/vendors/wodotexteditor/Tools.js
@@ -50,9 +50,11 @@ define("webodf/editor/Tools", [
                 onToolDone = args.onToolDone,
                 loadOdtFile = args.loadOdtFile,
                 saveOdtFile = args.saveOdtFile,
+                downloadOdtFile = args.downloadOdtFile,
                 close = args.close,
                 toolbar,
                 loadButton, saveButton, closeButton, aboutButton,
+                downloadButton,
                 formatDropDownMenu, formatMenuButton,
                 paragraphStylesMenuItem, paragraphStylesDialog, simpleStyles, currentStyle,
                 zoomSlider,
@@ -210,6 +212,22 @@ define("webodf/editor/Tools", [
                     saveButton.placeAt(toolbar);
                 }
 
+                // Download
+                if (downloadOdtFile) {
+                    downloadButton = new Button({
+                        label: tr('Download'),
+                        showLabel: true,
+                        style: {
+                            float: 'right'
+                        },
+                        onClick: function () {
+                            downloadOdtFile();
+                            onToolDone();
+                        }
+                    });
+                    downloadButton.placeAt(toolbar);
+                }
+
                 // Format menu
                 if (args.paragraphStyleEditingEnabled) {
                     formatDropDownMenu = new DropDownMenu({});
diff --git a/vendors/wodotexteditor/wodotexteditor.js b/vendors/wodotexteditor/wodotexteditor.js
index d5ac887..d4b0298 100644
--- a/vendors/wodotexteditor/wodotexteditor.js
+++ b/vendors/wodotexteditor/wodotexteditor.js
@@ -280,6 +280,7 @@ var Wodo = Wodo || (function () {
             //
             loadOdtFile = editorOptions.loadCallback,
             saveOdtFile = editorOptions.saveCallback,
+            downloadOdtFile = editorOptions.downloadCallback,
             close =       editorOptions.closeCallback,
             //
             directTextStylingEnabled = isEnabled(editorOptions.directTextStylingEnabled),
@@ -654,6 +655,7 @@ var Wodo = Wodo || (function () {
                 onToolDone: setFocusToOdfCanvas,
                 loadOdtFile: loadOdtFile,
                 saveOdtFile: saveOdtFile,
+                downloadOdtFile: downloadOdtFile,
                 close: close,
                 directTextStylingEnabled: directTextStylingEnabled,
                 directParagraphStylingEnabled: directParagraphStylingEnabled,
